### ISSUE TITLE
Use a Buffered Writer

### ISSUE THREAD

**Issue by adrian4096** [Arnau478/hevi#45]
![image](https://github.com/Arnau478/hevi/assets/146006794/3d8962c2-7bad-4e29-8ab9-b7bc7569fe88)
The easiest way to prevent this might just be with a defer statement, tho better ways of handling <Ctrl-C> exist

**Comment by adrian4096** [comment]
The above can be reproduced by running `hevi /dev/random` and clicking Ctrl-C

**Comment by adrian4096** [comment]
An even better fix would be to make line writes atomic. Linux supports this with syscalls. I am unaware of windows details but I suspect the same is probably true. In practice:
In the function at https://github.com/Arnau478/hevi/blob/e31c09bfc48ee9ed39436e1c689c03e86e7cc0fb/src/hevi.zig#L377 wrap the writer in: https://ziglang.org/documentation/0.13.0/std/#std.io.BufferedWriter and call flush after every line is done.

**Comment by Arnau478** [comment]
If I understand correctly what you mean, this is already implemented, take a look at:
https://github.com/Arnau478/hevi/blob/e31c09bfc48ee9ed39436e1c689c03e86e7cc0fb/src/hevi.zig#L314-L315

Also, how are you getting that output with `/dev/random`? It shouldn't work with `/dev/random`, as it first needs to read the whole file (to be able to detect the parser)

**Comment by adrian4096** [comment]
I am using the AUR version
Sorry for missing the buffered Writer. I had suspected, that this was the issue, due to incorrectly stopping mid output.
Is it really necessary to parse the whole file, before outputing?

**Comment by adrian4096** [comment]
pacman states, that my current version is `0.2.0-1`

**Comment by adrian4096** [comment]
One last thought, you could test for special files:
```
file /dev/urandom 
/dev/urandom: character special (1/9)
```
So that you can avoid finding the end of file for infinite files.
Or if infinite files are considered out of scope that's ok too.

**Comment by Arnau478** [comment]
Well, under linux, we could check the VFS node type and, if it's a chardev, disable parsing... However, I don't see a real use case for it. I assume you were just doing `hevi /dev/urandom` to easily get the lines truncated, right? If I'm missing some actual use case for reading endless files please tell me or open an issue, I'll be happy to consider it then :)

**Comment by adrian4096** [comment]
Correct, tho ideologically, I would prefer hexdump not to crash on any input file, I see your reasons and find your decision sensible.

**Comment by Arnau478** [comment]
> Correct, tho ideologically, I would prefer hexdump not to crash on any input file, I see your reasons and find your decision sensible.

`hevi` doesn't crash, it just keeps reading waiting to find an EOF

**Comment by adrian4096** [comment]
The halt and do nothing seems worse or equal to a crash. You can't start outputing lines, until you identify the file right?
